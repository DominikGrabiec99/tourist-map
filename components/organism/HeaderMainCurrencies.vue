<template>
  <div
    class="currencies-header transition-color bg-gray-50 text-slate-900 dark:bg-neutral-900 dark:text-gray-200 py-1 relative"
    :class="{'cursor-pointer': !showCurrencyView}"
    @click.self="$emit('toggle-currency-view', true)"
  >
    <div
      class="currencies-box flex whitespace-nowrap container mx-auto px-4 md:px-1 lg:px-20 overflow-hidden"
      :class="{ hidden: !showCurrencyView }"
      @click="$emit('toggle-currency-view', true)"
    >
      <OrganismMainCurrencies :currencies="mainCurrencies" />
      <OrganismMainCurrencies :currencies="mainCurrencies" />
    </div>
    <Icon
      v-if="showCurrencyView"
      name="material-symbols:keyboard-double-arrow-up-rounded"
      size="20"
      id="currency-toggle-icon"
      class="absolute dark:text-white text-neutral-900 bg-white dark:bg-neutral-900 rounded-full bottom-0 left-1/2 -translate-x-1/2 translate-y-1/2 z-50 cursor-pointer !hidden"
      @click="$emit('toggle-currency-view', false)"
    />
  </div>
</template>

<script setup lang="ts">
/** INTERFACES */
import IMainCurrency from '@/ts/interfaces/MainCurrency';

const { mainCurrencies = [] } = defineProps<{
  mainCurrencies: IMainCurrency[];
  showCurrencyView: boolean;
}>();

defineEmits<{
  (e: 'toggle-currency-view', value: boolean): void;
}>();
</script>

<style lang="scss" scoped>
.currencies-header:hover #currency-toggle-icon {
  display: block !important;
}

.currencies-header:hover .currencies-box {
  display: flex;
}
</style>
