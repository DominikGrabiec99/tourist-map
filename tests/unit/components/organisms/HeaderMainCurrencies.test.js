import { describe, it } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import HeaderMainCurrencies from '@/components/organism/HeaderMainCurrencies.vue';

import { mainCurrencies } from '@/tests/mockData';

describe('HeaderMainCurrencies.vue', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(HeaderMainCurrencies, {
      props: {
        mainCurrencies,
        showCurrencyView: true,
      },
    });
  });

  it('render component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('check toggle emitted currency view', async () => {
    wrapper.vm.$emit('toggle-currency-view', !wrapper.vm.showCurrencyView);
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted('toggle-currency-view')[0]).toEqual([false]);

    wrapper.find('#currency-toggle-icon').trigger('click');
    expect(wrapper.emitted('click')).toHaveLength(1);

    wrapper.find('.currencies-box').trigger('click');
    expect(wrapper.emitted('click')).toHaveLength(2);
  });
});
