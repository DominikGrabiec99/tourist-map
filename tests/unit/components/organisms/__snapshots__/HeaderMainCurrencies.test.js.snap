// Vitest Snapshot v1

exports[`HeaderMainCurrencies.vue > render component 1`] = `
<div
  class="currencies-header transition-color bg-gray-50 text-slate-900 dark:bg-neutral-900 dark:text-gray-200 py-1 relative"
  data-v-24e61421=""
>
  <div
    class="currencies-box flex whitespace-nowrap container mx-auto px-4 md:px-1 lg:px-20 overflow-hidden"
    data-v-24e61421=""
  >
    <organismmaincurrencies
      currencies="[object Object],[object Object]"
      data-v-24e61421=""
    />
    <organismmaincurrencies
      currencies="[object Object],[object Object]"
      data-v-24e61421=""
    />
  </div>
  <icon
    class="absolute dark:text-white text-neutral-900 bg-white dark:bg-neutral-900 rounded-full bottom-0 left-1/2 translate-x-1/2 translate-y-1/2 z-50 cursor-pointer !hidden"
    data-v-24e61421=""
    id="currency-toggle-icon"
    name="system-uicons:arrow-up-circle"
    size="20"
  />
</div>
`;
